### PR TITLE
Resources: Add Featured Articles slider

### PR DESCRIPTION
**Pull request by ladytrekker**
## Description

This is the next part of the new Resources page:

- build new slider for Resources Page
- query posts with flag "isFeaturedArticle"
- pass these articles to Slider

## Related Ticket

Part of https://github.com/KlimaDAO/klimadao/issues/534

## Preview

See https://klimadao-site-git-resources-featured-articles-slider-klimadao.vercel.app/resources

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

File: site/components/pages/Resources/FeaturedArticles/ArticlesSlider.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  articles: FeaturedPost[];
+};
+
+export const ArticlesSlider: FC<Props> = (props) => {
+  const articleRef = useRef(null);
+  const sliderRef = useRef<HTMLDivElement>(null);
+
+  const [articleWidth] = useElementWidth(articleRef);
+  const [sliderWidth] = useElementWidth(sliderRef);
+  const [, setScrollInterval] = useState<NodeJS.Timeout>();
```
Comment (Atmosfearful) [inline site/components/pages/Resources/FeaturedArticles/ArticlesSlider.tsx:21]: I think `useRef` is a better fit for this, plus then it doesn't cause re-renders (unless you want re-render?)

File: site/hooks/useElementWidth/index.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  useEffect(() => {
+    firstUpdateWidth();
+    window.addEventListener("resize", updateWidth);
+    return () => {
+      window.removeEventListener("resize", updateWidth);
+    };
+  }, [updateWidth]);
+
+  return [width];
```
Comment (Atmosfearful) [inline site/hooks/useElementWidth/index.ts:41]: would be neat if you also manage the ref inside this hook, so the parent doesnt need to pass anything in
```suggestion
  return [width, ref];
```

```
const [articleWidth, articleRef] = useElementWidth()
```

File: site/components/pages/Resources/index.tsx [hunk 3]
```diff
@@ -9,14 +9,23 @@ import { Navigation } from "components/Navigation";
...
+import { Document, FeaturedPost } from "lib/queries";
 
 export interface Props {
   documents: Document[];
+  featuredArticles: FeaturedPost[];
 }
 
-export const Resources: NextPage<Props> = ({ documents }) => {
+export const Resources: NextPage<Props> = ({ documents, featuredArticles }) => {
+  console.log("WORKS", featuredArticles);
```
Comment (Atmosfearful) [inline site/components/pages/Resources/index.tsx:22]: loggies

File: site/hooks/useElementWidth/index.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      setWidth(width);
+    }
+  }, [elementRef]);
+
+  const firstUpdateWidth = useCallback(() => {
+    if (elementRef && elementRef.current) {
+      let { width } = elementRef.current.getBoundingClientRect();
+      width =
+        width -
+        parseInt(
```
Comment (Atmosfearful) [inline site/hooks/useElementWidth/index.ts:18]: just a personal preference:

```suggestion
      const rect = elementRef.current.getBoundingClientRect();
      const width =
        rect.width -
        parseInt(
```

File: site/lib/queries.ts [hunk 5]
```diff
@@ -94,6 +108,8 @@ export type PostDetails = {
   imageUrl?: string;
 };
 export type AllPosts = PostDetails[];
+export type FeaturedPost = PostDetails & { isFeaturedArticle: true };
```
Comment (Atmosfearful) [inline site/lib/queries.ts:111]: do we support featured Podcasts up there too? Might be worth supporting if it's not too hard.

**Review by Atmosfearful — APPROVED** [review]
smol things! looks great!
Might have design nitpicks but let's get some good covers (without overlapping text) so we can enjoy it in it's true glory first

File: site/components/pages/Resources/index.tsx (hunk 3 above)
Comment (ladytrekker) [reply site/components/pages/Resources/index.tsx:22]: Yes these were there for you 😄 

Removed now as "real" featured articles are queried now

File: site/lib/queries.ts (hunk 5 above)
Comment (ladytrekker) [reply site/lib/queries.ts:111]: Not according to any design or concept.
I added it to the "Open UI questions" lists here https://github.com/KlimaDAO/klimadao/issues/534

File: site/hooks/useElementWidth/index.ts (hunk 2 above)
Comment (ladytrekker) [reply site/hooks/useElementWidth/index.ts:41]: The hook is used for the infinity slider too.
Will leave it as is right now and let´s do this in a follow-up

File: site/components/pages/Resources/FeaturedArticles/ArticlesSlider.tsx (hunk 1 above)
Comment (ladytrekker) [reply site/components/pages/Resources/FeaturedArticles/ArticlesSlider.tsx]: Done

**Pull request merged**